jwst/pipeline/calwebb_spec2.py

```diff
@@ -237,8 +238,26 @@ def process_exposure_product(
                         raise RuntimeError('Cannot determine WCS.')
 
         # Steps whose order is the same for all types of input.
+
+        # Leakcal subtraction (imprint)  occurs before background subtraction on a per-exposure basis.
+        # If there is only one `imprint` member, this imprint exposure is subtracted from all the
+        # science and background exposures.  Otherwise, there will be as many `imprint` members as
+        # there are science plus background members. Which imprint to subtract from which science/background
+        # is determined by the matching values of the header keyword PATT_NUM, or
+        # `DataModel.meta.dither.position_number`.
+
+        # find the position number of all the leak cal observations
+        imprint_pos_no = self._imprint_pos_no(members_by_type)
+
+        calibrated = self.imprint_subtract(calibrated, members_by_type['imprint'], imprint_pos_no)
```

Comment:
I think I agree with @stscieisenhamer that all of the functionality associated with figuring out dither position numbers and linking leakcals to other members should be contained within the `imprint` step itself, rather than adding all of that boiler plate code up here in the pipeline module. That would also mean that the argument list to `imprint_subtract` could remain simpler, in that it should only need to be sent `calibrated` and `members_by_type`. The population of the `imprint_pos_no` list could be done down within the step, as opposed to constructing it up here and then passing it down. It isn't needed anywhere in the pipeline module itself, only in the `imprint` step.